code/datums/components/food/edible.dm

```diff
@@ -75,7 +75,7 @@ Behavior that's still missing from this component that original food items had t
 		if (!item.grind_results)
 			item.grind_results = list() //If this doesn't already exist, add it as an empty list. This is needed for the grinder to accept it.
 
-	else if(isturf(parent))
+	else if(isturf(parent) || isstructure(parent))
 		RegisterSignal(parent, COMSIG_ATOM_ATTACK_HAND, .proc/TryToEatTurf)
```

Comment:
`TryToEatTurf` is an outdated name now, should be renamed to something else.

Reply:
Also, test this with every one of the custom mat item pls.